### PR TITLE
Add seeding for Record, Certificate, and Challenge

### PR DESCRIPTION
**Pull request by cychu42**
Closes: #143 
Let me know if you think more example data should be included, as well as whether the examples need changes.

## Changes
- Code for seeding Record, Certificate, and Challenge tables in `prisma/seed.ts`.
- Code for cleaning all existing data in the database in `prisma/seed.ts`.

## Testing
1. Set up the project normally to see if seeding works
or
2. run `npx prisma db seed` in a working copy of the project to test it

### CODE REVIEW

File: prisma/seed.ts [hunk 1]
```diff
@@ -23,6 +25,118 @@ async function seed() {
     },
   });
 
+  await prisma.record.createMany({
+    data: [
+      // A record
+      {
+        id: 1,
```
Comment (humphd) [inline prisma/seed.ts:32]: Based on https://github.com/Seneca-CDOT/starchart/blob/main/prisma/schema.prisma#L23, you shouldn't need to provide an `id`, it should automatically assign a new one.

File: prisma/seed.ts [hunk 2]
```diff
@@ -23,6 +25,118 @@ async function seed() {
...
+        username: username,
+        name: 'OSD700',
+        type: `A`,
+        value: '172.5.251.0',
+        description: 'For final porject.',
```
Comment (humphd) [inline prisma/seed.ts:37]: Typo in `project`

File: prisma/seed.ts [hunk 3]
```diff
@@ -23,6 +25,118 @@ async function seed() {
     },
   });
 
+  await prisma.record.createMany({
+    data: [
+      // A record
+      {
+        id: 1,
+        username: username,
+        name: 'OSD700',
```
Comment (humphd) [inline prisma/seed.ts:34]: Prefer lowercase for all `name` values`

File: prisma/seed.ts [hunk 4]
```diff
@@ -23,6 +25,118 @@ async function seed() {
     },
   });
 
+  await prisma.record.createMany({
+    data: [
+      // A record
+      {
+        id: 1,
+        username: username,
```
Comment (humphd) [inline prisma/seed.ts:33]: When the key/value are the same, list it once: `username,`

File: prisma/seed.ts [hunk 5]
```diff
@@ -23,6 +25,118 @@ async function seed() {
...
+        createdAt: new Date('2022-03-15T01:01:01'),
+        expiresAt: new Date('2022-09-15T12:59:59'),
+        status: `pending`,
+      },
+    ],
+  });
+
+  await prisma.certificate.create({
+    data: {
+      id: 1,
```
Comment (humphd) [inline prisma/seed.ts:110]: Same comments here.  Let Prisma do the automatic fields for you.

**Review by humphd — CHANGES_REQUESTED** [review]

File: prisma/seed.ts (hunk 2 above)
Comment (cychu42) [reply prisma/seed.ts:37]: Ah, thanks.

File: prisma/seed.ts (hunk 4 above)
Comment (cychu42) [reply prisma/seed.ts:33]: I see. Thank you.

File: prisma/seed.ts (hunk 1 above)
Comment (humphd) [reply prisma/seed.ts:32]: Based on the call today, sounds like you want this for foreign key.  Resolving.

File: prisma/seed.ts [hunk 6]
```diff
@@ -23,6 +25,118 @@ async function seed() {
...
+      username: username,
+      subject: `CN=*.${username}.example.com`,
+      certificate:
+        '-----BEGIN CERTIFICATE-----ApfFCv0O65TYkp5jEWSlU8PhKYD43nXA=-----END CERTIFICATE-----',
+      orderUrl: `orderURL.example.com`,
+      privkey:
```
Comment (SerpentBytes) [inline prisma/seed.ts:116]: Maybe follow camelCase convention here to be consistent with the rest of the code.

Comment (cychu42) [reply prisma/seed.ts:113]: OK, done.

**Comment by cychu42** [comment]
Thank you for the feedbacks. I did a number of changes:
- I would like to keep the specified `id`s to avoid situations where there’s is an issue referencing them for foreign keys. Otherwise. auto incrementing will just keep incrementing the number every time someone runs the seeding command.
- For dates, I changed dates of creation to be now. Expiration dates are a certain time from now. 6 months for records, and 90 days for certificates.
- The challenge part is removed as Denes suggested.
- `orderUrl` is also changed as suggested.

**Comment by SerpentBytes** [comment]
@cychu42 is privkey supposed to be privKey?

**Review by humphd — CHANGES_REQUESTED** [review]
Looks good, other than the `privKey` vs. `privkey` change @SerpentBytes wants.

**Comment by cychu42** [comment]
> @cychu42 is privkey supposed to be privKey?

It's supposed to be `privkey`, as it's what's in the schema. However, I see that it should probably be `privKey` or even `privateKey`. I originally took it from the [Data Architecture](https://github.com/Seneca-CDOT/starchart/wiki/Data-Architecture) page when making the schema.
I think the later (`privateKey`) makes more sense as it will be clearer to new readers.

If we are going to change it, I can add it as a suggestion under another PR to change the schema, like #153. Just to keep scope of issues separate.

**Review by humphd — APPROVED** [review]

**Review by dadolhay — APPROVED** [review]

**Review by SerpentBytes — APPROVED** [review]

**Pull request merged**